### PR TITLE
packages: title localization fix

### PR DESCRIPTION
**Pull request by FernandoAscencio**
#### What it does

Closes https://github.com/eclipse-theia/theia/issues/11954#issue-1476836525.

The commit fixes localization bugs found in the titles of windows.

While in the process of solving this issue, a check on other title variables found several other windows with similar localization issues. In total six issues should be fixed.

#### How to test

To test if the initial issue was solved:
1. start the application
2. open the debug-view (ex: `view` > `debug)`
3. from the debug-view in the `watch` section click `add expression` (+ icon)
4. a dialog will appear with the title `edit watch expression`
5. use the command `configure display language` (using `F1`) and install a new language
6. restart the application when prompted
7. repeat the steps 1-4 and confirm that the title of the dialog is localized

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- [x] As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: FernandoAscencio <fernando.ascencio.cama@ericsson.com>

### CODE REVIEW

File: packages/filesystem/src/browser/file-tree/file-tree-model.ts [hunk 1]
```diff
@@ -200,10 +201,10 @@ export class FileTreeModel extends CompressedTreeModel implements LocationServic
...
     protected async shouldReplace(fileName: string): Promise<boolean> {
         const dialog = new ConfirmDialog({
-            title: 'Replace file',
-            msg: `File '${fileName}' already exists in the destination folder. Do you want to replace it?`,
-            ok: 'Yes',
-            cancel: 'No'
+            title: nls.localize('theia/filesystem/replaceTitle', 'Replace file'),
+            msg: nls.localize('theia/filesystem/fileVariableInput', 'File \'{0}\' already exists in the destination folder. Do you want to replace it?', fileName),
+            ok: nls.localizeByDefault('Yes'),
+            cancel: nls.localizeByDefault('No')
```
Comment (vince-fugnitto) [inline packages/filesystem/src/browser/file-tree/file-tree-model.ts:207]: I believe we can use the exported constants which will already be localized:

```suggestion
            ok: Dialog.Yes,
            cancel: Dialog.No
```

File: packages/filesystem/src/browser/file-tree/file-tree-model.ts [hunk 2]
```diff
@@ -200,10 +201,10 @@ export class FileTreeModel extends CompressedTreeModel implements LocationServic
 
     protected async shouldReplace(fileName: string): Promise<boolean> {
         const dialog = new ConfirmDialog({
-            title: 'Replace file',
-            msg: `File '${fileName}' already exists in the destination folder. Do you want to replace it?`,
-            ok: 'Yes',
-            cancel: 'No'
+            title: nls.localize('theia/filesystem/replaceTitle', 'Replace file'),
+            msg: nls.localize('theia/filesystem/fileVariableInput', 'File \'{0}\' already exists in the destination folder. Do you want to replace it?', fileName),
```
Comment (vince-fugnitto) [inline packages/filesystem/src/browser/file-tree/file-tree-model.ts:205]: I believe there is a default we can use:

```suggestion
            msg: nls.localizeByDefault('{0} already exists. Are you sure you want to overwrite it?', fileName)
```

File: packages/git/src/browser/git-scm-provider.ts [hunk 3]
```diff
@@ -415,14 +415,14 @@ export class GitScmProvider implements ScmProvider {
             fileText = `${paths.length} files`;
         }
         return new ConfirmDialog({
-            title: 'Discard changes',
+            title: nls.localize('theia/git/replaceConfirmDiscardTitle', 'Discard changes'),
```
Comment (vince-fugnitto) [inline packages/git/src/browser/git-scm-provider.ts:418]: I believe that the following is already used, and can be re-used:

```suggestion
            title: nls.localize('vscode.git/package/command.clean', 'Discard changes'),
```

File: packages/plugin-ext/src/common/plugin-api-rpc.ts [hunk 4]
```diff
@@ -2036,7 +2036,7 @@ export interface TabsExt {
 export interface TabsMain {
     $moveTab(tabId: string, index: number, viewColumn: EditorGroupColumn, preserveFocus?: boolean): void;
     $closeTab(tabIds: string[], preserveFocus?: boolean): Promise<boolean>;
-    $closeGroup(groupIds: number[], preservceFocus?: boolean): Promise<boolean>;
+    $closeGroup(groupIds: number[], preserveFocus?: boolean): Promise<boolean>;
```
Comment (vince-fugnitto) [inline packages/plugin-ext/src/common/plugin-api-rpc.ts:2039]: I believe this is unrelated and part of https://github.com/eclipse-theia/theia/pull/12061.

File: packages/filesystem/src/browser/file-tree/file-tree-model.ts [hunk 5]
```diff
@@ -200,10 +201,10 @@ export class FileTreeModel extends CompressedTreeModel implements LocationServic
 
     protected async shouldReplace(fileName: string): Promise<boolean> {
         const dialog = new ConfirmDialog({
-            title: 'Replace file',
-            msg: `File '${fileName}' already exists in the destination folder. Do you want to replace it?`,
-            ok: 'Yes',
-            cancel: 'No'
+            title: nls.localize('theia/filesystem/replaceTitle', 'Replace file'),
```
Comment (vince-fugnitto) [inline packages/filesystem/src/browser/file-tree/file-tree-model.ts:204]: minor: I think the following would be a better title overall

```suggestion
            title: nls.localize('theia/filesystem/replaceTitle', 'Replace File'),
```

**Review by vince-fugnitto — APPROVED** [review]
The changes look good to me :+1:

**Review by msujew — APPROVED** [review]
Looks good to me as well 👍

File: packages/git/src/browser/git-scm-provider.ts [hunk 6]
```diff
@@ -415,14 +415,14 @@ export class GitScmProvider implements ScmProvider {
             fileText = `${paths.length} files`;
         }
         return new ConfirmDialog({
-            title: 'Discard changes',
+            title: nls.localize('vscode.git/package/command.clean', 'Discard changes'),
```
Comment (vince-fugnitto) [inline packages/git/src/browser/git-scm-provider.ts:418]: minor: I hadn't noticed before but I believe we should update to the following:

```suggestion
            title: nls.localize('vscode.git/package/command.clean', 'Discard Changes'),
```
